package.json

```diff
@@ -1205,6 +1205,143 @@
           ],
           "scope": "window"
         },
+        "cmake.statusbar.type": {
+          "type": "string",
+          "default": "text",
+          "enum": [
+            "text",
+            "short",
+            "icon",
+            "hidden"
+          ],
+          "scope": "window"
+        },
+        "cmake.statusbar.advanced": {
+          "type": "object",
+          "default": null,
+          "properties": {
+            "kit": {
+              "type": "object",
+              "properties": {
+                "type": {
+                  "type": "string",
+                  "enum": [
+                    "text",
+                    "short",
+                    "icon",
+                    "hidden"
+                  ]
+                },
+                "length": {
+                  "type": "integer"
+                }
+              }
+            },
+            "status": {
+              "type": "object",
+              "properties": {
+                "type": {
+                  "type": "string",
+                  "default": "text",
+                  "enum": [
+                    "text",
```

Comment:
I'm wondering if the enum values should be the same for all of these settings (even if they don't have a visible effect). By having a single setting that configures all buttons, we are forced to interpret the values for each of the buttons already.

Reply:
i added the missing values.